File: SmartHomeAndroid-old/app/src/main/java/com/github/wangxuxin/smarthome/AboutActivity.java
package com.github.wangxuxin.smarthome;

import android.content.Intent;
import android.content.pm.PackageInfo;
import android.content.pm.PackageManager;
import android.net.Uri;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.util.Log;
import android.view.View;
import android.widget.TextView;

/**
 * Created by a1274 on 2016/12/9.
 */
public class AboutActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_about);
        //ActionBar actionBar = getActionBar();
        //actionBar.setDisplayHomeAsUpEnabled(true);

        TextView versionText=(TextView)findViewById(R.id.textView_version);
        try {
            versionText.setText("V"+getVersionName());
        } catch (Exception e) {
            e.printStackTrace();
        }

        TextView copyrightText = (TextView) findViewById(R.id.textView_copyright);
        if(!copyrightText.getText().equals("By wxx")){
            copyrightText.setText("By wxx");
            Log.d("wxxDebugAbout","原程序已被修改！！！！By wxx");
        }

        TextView websiteText = (TextView) findViewById(R.id.textView_website);
        websiteText.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                String url = "https://wangxuxin.github.io"; // web address
                Intent intent = new Intent(Intent.ACTION_VIEW);
                intent.setData(Uri.parse(url));
                startActivity(intent);
            }
        });
    }

    /*
    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        switch (item.getItemId()) {
            case android.R.id.home:
                finish();
                return true;
            default:
                return super.onOptionsItemSelected(item);
        }
    }
    */

    public String getVersionName() throws Exception {
        // 获取packagemanager的实例
        PackageManager packageManager = getPackageManager();
        // getPackageName()是你当前类的包名，0代表是获取版本信息
        PackageInfo packInfo = packageManager.getPackageInfo(getPackageName(),0);
        Log.d("wxxDebugAbout",packInfo.versionName);
        return packInfo.versionName;
    }
}

File: SmartHomeAndroid/app/src/main/java/io/github/acedroidx/smarthome/AboutActivity.kt
package io.github.acedroidx.smarthome

import android.content.Intent
import android.content.pm.PackageManager
import android.net.Uri
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.widget.TextView

class AboutActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_about)

        val versionText = findViewById<TextView>(R.id.textView_version)
        try {
            versionText.text = "V$versionName"
        } catch (e: Exception) {
            e.printStackTrace()
        }

        val copyrightText = findViewById<TextView>(R.id.textView_copyright)
        if (copyrightText.text != "By AceDroidX") {
            copyrightText.text = "By AceDroidX"
            Log.d("wxxDebugAbout", "原程序已被修改！！！！By wxx")
        }

        val websiteText = findViewById<TextView>(R.id.textView_website)
        val appInfo = this.packageManager
            .getApplicationInfo(
                this.packageName,
                PackageManager.GET_META_DATA
            )
        val url = appInfo.metaData.getString("Website")
        websiteText.text = url
        websiteText.setOnClickListener {
            val intent = Intent(Intent.ACTION_VIEW)
            websiteText.text = url
            intent.data = Uri.parse(url)
            startActivity(intent)
        }
    }

    // 获取packagemanager的实例
    // getPackageName()是你当前类的包名，0代表是获取版本信息
    val versionName: String
        get() {
            val packageManager = packageManager
            val packInfo = packageManager?.getPackageInfo(packageName, 0)
            Log.d("wxxDebugAbout", packInfo!!.versionName)
            return packInfo.versionName
        }
}
